### DISCUSSION TITLE
是否可以像rclone一样兼容OneDrive商业版本身的WebDAV功能？

### DISCUSSION THREAD

**Discussion by ChirmyRam** [AlistGo/alist#460] [Ideas]
作者新年快乐。
OneDrive商业版本身不支持目前通行的WebDAV协议，但它确实有比较特殊的WebDAV功能，目前只有[rclone](https://github.com/rclone/rclone/)能比较完美的支持它。以我的E5 OneDrive登录后首页的地址为例：
```
https://chirmyram-my.sharepoint.com/personal/office_chirmyram_top/_layouts/15/onedrive.aspx
```
则其WebDAV URL为：
```
https://chirmyram-my.sharepoint.com/personal/office_chirmyram_top/Documents/
```
观察其特点可发现，将末尾的`/_layouts/15/onedrive.aspx`替换为`/Documents/`就可以了。末尾`/Documents/`即为OneDrive商业版根目录，也可在其后继续添加子目录。在[rclone](https://github.com/rclone/rclone/)里面配置的时候，选择储存类型为WebDAV -> 输入上述WebDAV URL -> 选择vendor为`Sharepoint Online, authenticated by Microsoft account`，直接输入OneDrive商业版账户密码即可配置成功。优点在于直接使用账号密码登录永久有效，直接绕过API配置、区域选择等参数，不必担心token过期。

[rclone](https://github.com/rclone/rclone/) WebDAV储存协议的源码在这个地方：[/backend/webdav](https://github.com/rclone/rclone/tree/master/backend/webdav)，alist同为用Go写的程序，我猜测移植过来会更容易一些。所以还请作者考虑一下，非常感谢。

**Comment by xhofe** [comment]
8e059c64b54246785e7b7ce833469e3cfe72e556

**Reply by ChirmyRam** [reply]
感谢。

**Reply by xhofe** [reply]
> 感谢。

似乎世纪互联不支持？我用rclone也挂不上

**Reply by ChirmyRam** [reply]
> > 感谢。
> 
> 似乎世纪互联不支持？我用rclone也挂不上

手里没有世纪互联的号，倒是没试过🤔

**Reply by SkysCrystal** [reply]
我的世纪互联 Onedrive 挂载不上
替换链接 `/Documents/` 后，显示 `错误 PROPFIND /: 403`。
按照Onedrive挂载，没有`和个人Microsoft账户...`这个选项

**Reply by ChirmyRam** [reply]
> 我的世纪互联 Onedrive 挂载不上 替换链接 `/Documents/` 后，显示 `错误 PROPFIND /: 403`。 按照Onedrive挂载，没有`和个人Microsoft账户...`这个选项

但是alist现在支持挂载世纪互联的WebDAV：https://alist-doc.nn.ci/docs/driver/webdav/ ，不清楚开发者的情况。个人OneDrive的WebDAV又比较特殊，跟商业版不一样。

**Reply by SkysCrystal** [reply]
我用的是教育邮箱的账号，挂载有点问题

**Reply by leekkk2** [reply]
> > 我的世纪互联 Onedrive 挂载不上 替换链接 `/Documents/` 后，显示 `错误 PROPFIND /: 403`。 按照Onedrive挂载，没有`和个人Microsoft账户...`这个选项
> 
> 但是alist现在支持挂载世纪互联的WebDAV：https://alist-doc.nn.ci/docs/driver/webdav/ ，不清楚开发者的情况。个人OneDrive的WebDAV又比较特殊，跟商业版不一样。

我现在挂载也是这个错误，楼主你是咋解决的呀

**Reply by SkysCrystal** [reply]
我隔了短时间又按流程走了一遍，就又可以了

**Comment by leekkk2** [comment]
这个好像还是不支持对吧？

**Reply by ChirmyRam** [reply]
早已支持 https://alist.nn.ci/zh/guide/drivers/webdav.html#onedrive-sharepoint

**Reply by leekkk2** [reply]
感谢回复，不知道是我哪里出错了，我按照教程去配置，报了403和楼上提到的错误一样

**Reply by SGaoNut** [reply]
不知道能不能分享下，配置的页面？有个人信息，可以打个码，挂载路径和地址分不清楚应该配哪个？

**Reply by ChirmyRam** [reply]
挂载路径是在alist首页显示的路径，填/或者/自定义名字。

**Reply by SGaoNut** [reply]
时间长了，看了其他的，搞明白了，但是配置了以后。
显示：failed get objs: failed to list objs: PROPFIND /: 403

**Reply by ChirmyRam** [reply]
多半链接填错了。
